lib/reporting/reports/enterprise_fee_summary/enterprise_fees_with_tax_report_by_producer.rb

```diff
@@ -0,0 +1,352 @@
+# frozen_string_literal: true
+
+module Reporting
+  module Reports
+    module EnterpriseFeeSummary
+      class EnterpriseFeesWithTaxReportByProducer < ReportTemplate
+        attr_accessor :permissions
+
+        def initialize(user, params = {}, render: false)
+          super(user, params, render: render)
+          @permissions = Permissions.new(user)
+        end
+
+        def search
+          report_line_items.orders
+        end
+
+        def order_permissions
+          return @order_permissions unless @order_permissions.nil?
+
+          @order_permissions = ::Permissions::Order.new(user, ransack_params)
+        end
+
+        def report_line_items
+          @report_line_items ||= Reporting::LineItems.new(order_permissions, params)
+        end
+
+        def supplier_ids_filter(supplier_id)
+          return true if params[:supplier_id_in].blank?
+
+          params[:supplier_id_in].include?(supplier_id)
+        end
```

Comment:
In Rails, methods returning a boolean value usually end with `?` to mark that. What do you think about this version? (just my opinion)

```suggestion
        def filter_supplier_ids?(supplier_id)
          params[:supplier_id_in].blank? ||
            params[:supplier_id_in].include?(supplier_id)
        end
```